src/main.sh

```diff
@@ -1,9 +1,29 @@
 #!/bin/bash
+source ./src/modules/yaml.sh
 
 echo "Install programs.."
 
-IFS=" " read -r -a programs <<< "$(find ./src/programs -type f | sed 's!.*/!!' | tr '\n' ' ')"
-sudo xbps-install -y "${programs[@]}"
+find ./src/programs -type f -print0 | while IFS= read -r -d $'\0' file; 
+do
+  package_name=$(basename $file)
+  base_name=$(echo $package_name | tr "-" "_") 
+  create_variables $file $base_name
+
+  preinstall="${base_name}preinstall"
+  install="${base_name}install"
+  postinstall="${base_name}postinstall"
+
+  [ -n "${!preinstall}" ] && eval $(echo ${!preinstall})
+
+  if [ -n "${!install}" ]; then
+    eval $(echo ${!install})
+  else
+    sudo xbps-install ${package_name} -y
+  fi
+
+  [ -n "${!postinstall}" ] && eval $(echo ${!postinstall})
```

Comment:
**[shellcheck]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub>
Double quote to prevent globbing and word splitting. [SC2086](https://github.com/koalaman/shellcheck/wiki/SC2086)